streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamImpl.java

```diff
@@ -685,13 +685,13 @@ public boolean test(final K1 key, final V1 value) {
                                                                                     windows.maintainMs());
 
             KStreamKStreamJoin<K1, R, V1, V2> joinThis = new KStreamKStreamJoin<>(otherWindow.name(),
-                                                                                  windows.before,
                                                                                   windows.after,
+                                                                                  windows.before,
```

Comment:
Shouldn't it actually be on `joinOther` where `windows.before` and `windows.after` are reversed? i.e., given:

`stream1.join(stream2, JoinWindows.of(0).after(1000))`

when a record arrives in `stream2` at time `1000` you want to check for any matching records in `stream1` from `0 - 1000`